File: frontend/src/app/api/requests/create-monster-request.ts
import {IActiveStatsModifier, IMonsterData} from '../shared';
import {CreateItemRequest} from './create-item-request';

export interface CreateMonsterRequest {
    name: string;
    fightId?: number;
    data: IMonsterData;
    modifiers?: IActiveStatsModifier[];
    items: CreateItemRequest[]
}

File: backend/src/Naheulbook.Requests/Requests/CreateMonsterRequest.cs
using Naheulbook.Shared.TransientModels;

namespace Naheulbook.Requests.Requests;

[PublicAPI]
public class CreateMonsterRequest
{
    public required string Name { get; set; }
    public int? FightId { get; set; }
    public MonsterData? Data { get; set; }
    public IList<ActiveStatsModifier>? Modifiers { get; set; }
    public required IList<CreateItemRequest> Items { get; set; }
}